packages/react/src/components/createInlineOverlayComponent.tsx

```diff
@@ -107,13 +114,38 @@ export const createInlineOverlayComponent = <PropType, ElementType>(
           this.setState({ isOpen: false });
         }
 
-        this.props.onDidDismiss && this.props.onDidDismiss(evt);
+        const didComponentUnmount = this.ref.current === null;
```

Comment:
Optional: Would it be better to remove the `didDismiss` and `willPresent` listeners in `componentWillUnmount` instead?

Reply:
Investigated this path, results in the previous error:
> react-dom.development.js:67 Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.

If I remove the `didComponentUnmount` check even when removing the event listeners in `componentWillUnmount`.